src/Parentheses.cpp

```diff
@@ -1,5 +1,37 @@
 #include "Parentheses.hpp"
+#include <vector>
+#include <algorithm>
 
-bool Parentheses::isBalanced(std::string s) {
-    return true;
+bool Parentheses::isBalanced(std::string s)
+{
+    std::vector<std::string> inputFalse
+    {
+        "}{",
+        ")(",
+        "][",
+    };
+    return std::none_of(inputFalse.begin(),
+                        inputFalse.end(),
+                        [&](const auto &negative)
+    {
+        return s.find(negative) != std::string::npos;
+    });
+}
+bool Parentheses::isBalancedAlgo(std::string& s)
+{
+    s.erase(std::remove(s.begin(), s.end(), ' '), s.end());
```

Comment:
to może lepiej nie przekazywać stringa s przez &, bo go modyfikujecie tutaj, a to mogłoby być niespodziewane zachowanie.